src/TimeoutMigrationTool/NHibernate/NHibernateDialectValidator.cs

```diff
@@ -0,0 +1,28 @@
+﻿namespace Particular.TimeoutMigrationTool.NHibernate
+{
+    using System;
+    using McMaster.Extensions.CommandLineUtils;
+    using McMaster.Extensions.CommandLineUtils.Validation;
+    using System.ComponentModel.DataAnnotations;
+
+    public class NHibernateDialectValidator : IOptionValidator
```

Comment:
This is nice, we should check if we can do the same for the ravenversion parameter. That is, if we keep raven 3 around.